kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -95,8 +122,42 @@ server {
             local utils = require('pl.utils')
             local table_utils   = require('table_utils')
 
-            data          = ngx.req.get_body_data()
-            tmp_file_name = string.match(data, 'id=(.*)&')
+            body_data                       = ngx.req.get_body_data()
+            tmp_file_name                   = string.match(body_data,'id=(.*)&')
+            sha1_uploaded_file              = string.match(body_data,'sha1=(.-)&')
+            sha1_uploaded_file_with_prefix  = "sha1:" .. sha1_uploaded_file
+            size_uploaded_file              = string.match(body_data,'size=(.-)&')
+
+            -- Now that the file has successfully finished uploading, we can check if the hashes of file are the same.
+            -- Accept the name from Concent-Upload-Path header. We assume that this header was validated by
+            -- the gatekeeper app and is safe to use.
+            client_sha1 = ngx.req.get_headers()["Concent-Checksum"]
+            if  sha1_uploaded_file_with_prefix ~= client_sha1 then
+                ngx.status = 400
+                ngx.header.content_type    = "application/json; charset=utf-8"
+                ngx.header["X-SHA1"]       = " "
+                ngx.header["X-Session-Id"] = " "
+                ngx.say('{"error_message": "The hash of file does not match", "error_code": "hash-not-match"}')
+                os.remove("/srv/storage/" .. tmp_file_name)
+                os.remove("/srv/storage/" .. tmp_file_name .. ".shactx")
+                return ngx.exit(400)
+           end
+
+
+           -- Now that the file has successfully finished uploading, we can check if the size of file are the same.
+           -- Accept the name from Concent-File-Size header. We assume that this header was validated by
+           -- the gatekeeper app and is safe to use.
+           client_size= ngx.req.get_headers()["Concent-File-Size"]
+           if  size_uploaded_file ~= client_size then
+                ngx.status = 400
+                ngx.header.content_type    = "application/json; charset=utf-8"
+                ngx.header["X-SHA1"]       = " "
+                ngx.header["X-Session-Id"] = " "
```

Comment:
Why a single space? Can't it be set to an empty string?

Reply:
With empty string not working, default header is return in response.

Reply:
Then please add a comment explaining it. It's not obvious.